recipes/newrelic/infrastructure/ohi/mongodb/debian.yml

```diff
@@ -62,24 +59,344 @@ preInstall:
+            echo "Mongo shell is required to run the newrelic install" # TODO: Prompt to install if not present?
+            exit 46
+          fi
+
+    mongo_pre_req:
+      cmds:
+        - |
+          if [ -f /tmp/mongo-role.js ]; then
+            rm -f /tmp/mongo-role.js 
+          fi
+          tee -a /tmp/mongo-role.js > /dev/null <<-"EOT"
+            use admin
+            db.createRole({ role: "listCollections", privileges: [{ resource: { db: "", collection: "" }, actions: [ "listCollections" ] }], roles: [] })
+          EOT
+          if [ -f /tmp/mongo-user.js ]; then
+            rm -f /tmp/mongo-user.js 
+          fi
+          tee -a /tmp/mongo-user.js > /dev/null <<-"EOT"
+            use admin
+            db.createUser({ user: "MONGO_MONITORING_USERNAME", pwd: "MONGO_MONITORING_PASSWORD", roles: [ "clusterMonitor", "listCollections" ] })
+          EOT
+      
+    assert_input:
+      cmds:
+        - |
+          NEW_RELIC_ASSUME_YES="{{.NEW_RELIC_ASSUME_YES}}" # TODO: Implement NEW_RELIC_ASSUME_YES
+          NR_CLI_DB_HOSTNAME="{{.NR_CLI_DB_HOSTNAME}}"
+          NR_CLI_DB_PORT="{{.NR_CLI_DB_PORT}}"
+          NR_CLI_DB_USERNAME="{{.NR_CLI_DB_USERNAME}}"
+          NR_CLI_DB_PASSWORD="{{.NR_CLI_DB_PASSWORD}}"
+          NR_CLI_DB_CLUSTERNAME="{{.NR_CLI_DB_CLUSTERNAME}}"
+          NR_CLI_DB_AUTH="{{.NR_CLI_DB_AUTH}}"
+          NR_CLI_SSL="{{.NR_CLI_SSL}}"
+          NR_CLI_CLIENT_CERT_FILE="{{.NR_CLI_CLIENT_CERT_FILE}}"
+          NR_CLI_CERT_AUTH_FILE="{{.NR_CLI_CERT_AUTH_FILE}}"
+
+          # MongoDB defaults
+          NR_CLI_DB_HOSTNAME=${NR_CLI_DB_HOSTNAME:-localhost}
+          NR_CLI_DB_PORT=${NR_CLI_DB_PORT:-27017}
+          NR_CLI_DB_CLUSTERNAME=${NR_CLI_DB_CLUSTERNAME:-mongocluster}
+          NR_CLI_DB_AUTH=${NR_CLI_DB_AUTH:-admin}
+
+          MONGO_MONITORING_USERNAME=newrelic
+          MONGO_MONITORING_PASSWORD=$(echo -n $(date +%s | sha256sum | base64 | head -c 16); echo "oO0$")
+
+          sed -i -e "s/MONGO_MONITORING_USERNAME/$MONGO_MONITORING_USERNAME/g" \
+                 -e "s/MONGO_MONITORING_PASSWORD/$MONGO_MONITORING_PASSWORD/g" \
+                 /tmp/mongo-user.js
+
+          # Setup integration's config file for mongodb without/with SCRAM authentication enabled cases
+          mkdir -p "/etc/newrelic-infra/integrations.d"
+          apt-get -o Acquire::Check-Valid-Until=false update -yq
+          apt-get install nri-mongodb -y
+          if [ -f /etc/newrelic-infra/integrations.d/mongodb-config.yml ]; then
+            rm /etc/newrelic-infra/integrations.d/mongodb-config.yml;
+          fi
+          tee /etc/newrelic-infra/integrations.d/mongodb-config.yml > /dev/null <<-"EOT"
+          integrations:
+            - name: nri-mongodb
+              env:
+                # The mongos to connect to
+                HOST: NR_CLI_DB_HOSTNAME
+                # The port the mongos is running on
+                PORT: NR_CLI_DB_PORT
```

Comment:
Can we validate the port before installing?

Reply:
Will add to my to do.